lambdas/scripts/dynamodb_migration_prmp_202.py

```diff
@@ -0,0 +1,158 @@
+
+
+class VersionMigration:
+    def __init__(self, environment: str, table_name: str, dry_run: bool = False):
+        self.environment = environment
+        self.table_name = table_name
+        self.dry_run = dry_run
+        self.logger = logging.getLogger("VersionMigration")
+        self.dynamo_service = DynamoDBService()
+
+        self.target_table = f"{self.environment}_{self.table_name}"
+
+    def main(
+            self, entries: Iterable[dict]
+    ) -> list[tuple[str, Callable[[dict], dict | None]]]:
+
+        """
+        Main entry point. Returns a list of update functions with labels.
+        Accepts a list of entries for Lambda-based execution, or scans the table if `entries` is None.
+        """
+        self.logger.info("Starting version migration")
+        self.logger.info(f"Target table: {self.target_table}")
+        self.logger.info(f"Dry run mode: {self.dry_run}")
+
+        if entries is None:
+            self.logger.info("No entries provided — scanning entire table.")
+            raise ValueError("Entries must be provided to main().")
+            # entries = list(self.dynamo_service.stream_whole_table(self.target_table))
+
+        # Return list of (label, update_fn) pairs
+        return [
+            ("Custodian", self.update_custodian_entry),
+            ("Status", self.update_status_entry),
+            ("DocumentSnomedCodeType", self.update_document_snomed_code_type_entry),
+            ("DocStatus", self.update_doc_status_entry),
+        ]
+
+    def process_entries(
+        self,
+        label: str,
+        entries: Iterable[dict],
+        update_fn: Callable[[dict], dict | None],
+    ):
+        self.logger.info(f"Running {label} migration")
+
+        for index, entry in enumerate(entries, start=1):
+            item_id = entry.get("ID")
+            self.logger.info(
+                f"[{label}] Processing item {index} (ID: {item_id})"
+            )
+
+            updated_fields = update_fn(entry)
+            if not updated_fields:
+                self.logger.debug(
+                    f"[{label}] Item {item_id} does not require update, skipping."
+                )
+                continue
+
+            if self.dry_run:
+                self.logger.info(
+                    f"[Dry Run] Would update item {item_id} with {updated_fields}"
+                )
+            else:
+                self.logger.info(f"Updating item {item_id} with {updated_fields}")
+                self.dynamo_service.update_item(
+                    table_name=self.target_table,
+                    key_pair={"ID": item_id},
+                    updated_fields=updated_fields,
+                )
+                self.logger.info(f"{label} migration completed.")
+
+
+    @staticmethod
+    def update_custodian_entry(entry: dict) -> dict | None:
+        custodian = entry.get("Custodian")
+        current_gp_ods = entry.get("CurrentGpOds")
+
+        if not custodian and current_gp_ods:
```

Comment:
Think we might need to change this.

If the `custodian` is missing, take `current_gp_ods`
If `custodian` is present but differs from `current_gp_ods` take `current_gp_ods`
If `custodian` is the same as `current_gp_ods` we can skip

Check with @bethany-kish-nhs on this one, I think this comes back to the migration of the DECE, SUSP and REST data in CurrentGpODS

Reply:
@bethany-kish-nhs can you have a look at this?